lib/src/adder.dart

```diff
@@ -21,6 +21,12 @@ abstract class Adder extends Module {
   @protected
   late final Logic b;
 
+  /// The addition results [out].
+  Logic get out => output('out');
```

Comment:
I'm struggling with this pattern a bit because there are multiple output functions and they need to pull from the same logic computation.  So I try to store the output logic in fields that these functions can then return.  But I in order to declare those fields in the class, apparently I need to initialize them despite declaring as late...

`class RippleCarryAdder extends Adder {
  late final Logic _sum;
  late final Logic _carry;

  /// Constructs an n-bit adder based on inputs List of inputs.
  RippleCarryAdder(super.a, super.b,
      {super.name = 'ripple_carry_adder', Logic? carry}) {
    final sumList = <Logic>[];
    for (var i = 0; i < a.width; i++) {
      final fullAdder = FullAdder(a: a[i], b: b[i], carryIn: carry ?? Const(0));

      carry = fullAdder.carryOut;
      sumList.add(fullAdder.sum);
    }

    sumList.add(carry!);

    _sum <= sumList.rswizzle();
    _carry <= sum[sum.width - 1];
  }
  @override
  @protected
  Logic calculateSum() => _sum;

  @override
  @protected
  Logic calculateCarry() => _carry;
}`